dvc/remote/base.py

```diff
@@ -86,11 +86,7 @@ def __init__(self, repo, config):
         self.repo = repo
 
         self._check_requires(config)
-
-        core = config.get(Config.SECTION_CORE, {})
-        self.checksum_jobs = core.get(
-            Config.SECTION_CORE_CHECKSUM_JOBS, self.CHECKSUM_JOBS
-        )
+        self.checksum_jobs = self._get_checksum_jobs(repo, config)
```

Comment:
No need to pass `repo`, you have `self.repo` set already.

Reply:
The reason why I decided to pass **repo** as argument for avoiding initialization dependency - otherwise needs to care that `self.repo` is initialized above. But I'm not sure that it's a good way

Reply:
Sorry, not sure I follow. Could you elaborate/rephrase, please?

Reply:
I meant that needs first init `self.repo` like
```
def __init__(...):
   self.repo = ...
   self.checksum_jobs = ...
```
introduce complexity that `self.checksum_jobs` should be initialized after `self.repo`, so in terms of refactoring the order can change and it breaks the constructor

Reply:
Well, but it will break in an obvious way with an attribute error, as `self.repo` won't be set. I would understand your logic if you would've made _get_checksum_jobs a static method or a helper function, but you've made it a full-blown method so using `self.repo` there would be totally fine. But current implementation works too. I guess another way to make it obvious would be to use `_get_checksum_jobs(repo.config if repo else {}, config)` 🙂But, again, these are non-vital details and neat-picking, so your call 🙂 Current implementation for this is mergeable.